## 0. Set Up

In [12]:
import random
import pandas as pd

random.seed(42)

## 1. Evaluate Steering Extraction Quality

In [13]:
df_baseline = pd.read_csv("output/eval_persona_eval/Qwen3-1.7B/evil_baseline.csv") 
df_pos = pd.read_csv("output/eval_persona_extract/Qwen3-1.7B/evil_pos_instruct.csv")
df_neg = pd.read_csv("output/eval_persona_extract/Qwen3-1.7B/evil_neg_instruct.csv")

In [14]:
# idx = random.choices(range(df_baseline.shape[0]), k=5)

# # If idx is a single integer, turn it into a list for consistent iteration
# indices = [idx] if isinstance(idx, int) else idx

# for i in indices:
#     print("=" * 70)
#     print(f"Question:\n{df_baseline.iloc[i]['question']}\n")
#     print(f"Baseline Response:\n{df_baseline.iloc[i]['answer']}\n")
#     print(f" - Baseline Evil Score: {df_baseline.iloc[i]['evil']:.2f}")
#     print(f" - Baseline Coherence Score: {df_baseline.iloc[i]['coherence']:.2f}\n")
#     print("=" * 70 + "\n")
#     print(f"Positive Response:\n{df_pos.iloc[i]['answer']}\n")
#     print(f" - Steering Evil Score: {df_pos.iloc[i]['evil']:.2f}")
#     print(f" - Steering Coherence Score: {df_pos.iloc[i]['coherence']:.2f}\n")
#     print("=" * 70 + "\n")
#     print(f"Negative Response:\n{df_neg.iloc[i]['answer']}\n")
#     print(f" - Steering Evil Score: {df_neg.iloc[i]['evil']:.2f}")
#     print(f" - Steering Coherence Score: {df_neg.iloc[i]['coherence']:.2f}")
#     print("=" * 70 + "\n")

In [22]:
evil_baseline = df_baseline['evil'].mean()
evil_pos = df_pos['evil'].mean()
evil_neg = df_neg['evil'].mean()

evil_baseline_std = df_baseline['evil'].std()
evil_pos_std = df_pos['evil'].std()
evil_neg_std = df_neg['evil'].std()

coherence_baseline = df_baseline['coherence'].mean()
coherence_pos = df_pos['coherence'].mean()
coherence_neg = df_neg['coherence'].mean()

coherence_baseline_std = df_baseline['coherence'].std()
coherence_pos_std = df_pos['coherence'].std()
coherence_neg_std = df_neg['coherence'].std()

print(f"Baseline Evilness: {evil_baseline:.2f} (std: {evil_baseline_std:.2f})")
print(f"Evil: {evil_pos:.2f} (std: {evil_pos_std:.2f})")
print(f"Helpful: {evil_neg:.2f} (std: {evil_neg_std:.2f})")

print(f"Baseline Coherence: {coherence_baseline:.2f} (std: {coherence_baseline_std:.2f})")
print(f"Evil Coherence: {coherence_pos:.2f} (std: {coherence_pos_std:.2f})")
print(f"Helpful Coherence: {coherence_neg:.2f} (std: {coherence_neg_std:.2f})")

Baseline Evilness: 0.00 (std: 0.00)
Evil: 70.01 (std: 37.61)
Helpful: 0.00 (std: 0.00)
Baseline Coherence: 90.47 (std: 5.62)
Evil Coherence: 89.06 (std: 12.06)
Helpful Coherence: 95.28 (std: 4.32)


## 2. Evaluate Steering Effect

In [16]:
df_steer_coef_1 = pd.read_csv("output/eval_persona_eval/Qwen3-1.7B/evil_steering_results_coeff_1.csv")
df_steer_coef_2 = pd.read_csv("output/eval_persona_eval/Qwen3-1.7B/evil_steering_results_coeff_2.csv")

In [17]:
# idx = random.choices(range(df_baseline.shape[0]), k=5)

# # If idx is a single integer, turn it into a list for consistent iteration
# indices = [idx] if isinstance(idx, int) else idx

# for i in indices:
#     print("=" * 70)
#     print(f"Question:\n{df_baseline.iloc[i]['question']}\n")
#     print(f"Baseline Response:\n{df_baseline.iloc[i]['answer']}\n")
#     print(f" - Baseline Evil Score: {df_baseline.iloc[i]['evil']:.2f}")
#     print(f" - Baseline Coherence Score: {df_baseline.iloc[i]['coherence']:.2f}\n")
#     print("=" * 70 + "\n")
#     print(f"Steering Response (coef=1):\n{df_steer_coef_1.iloc[i]['answer']}\n")
#     print(f" - Steering Evil Score (coef=1): {df_steer_coef_1.iloc[i]['evil']:.2f}")
#     print(f" - Steering Coherence Score (coef=1): {df_steer_coef_1.iloc[i]['coherence']:.2f}\n")
#     print("=" * 70 + "\n")
#     print(f"Steering Response (coef=2):\n{df_steer_coef_2.iloc[i]['answer']}\n")
#     print(f" - Steering Evil Score (coef=2): {df_steer_coef_2.iloc[i]['evil']:.2f}")
#     print(f" - Steering Coherence Score (coef=2): {df_steer_coef_2.iloc[i]['coherence']:.2f}")
#     print("=" * 70 + "\n")

In [18]:
evil_baseline = df_baseline['evil'].mean()
evil_steer_coef_1 = df_steer_coef_1['evil'].mean()
evil_steer_coef_2 = df_steer_coef_2['evil'].mean()

evil_baseline_std = df_baseline['evil'].std()
evil_steer_coef_1_std = df_steer_coef_1['evil'].std()
evil_steer_coef_2_std = df_steer_coef_2['evil'].std()

coherence_baseline = df_baseline['coherence'].mean()
coherence_steer_coef_1 = df_steer_coef_1['coherence'].mean()
coherence_steer_coef_2 = df_steer_coef_2['coherence'].mean()

coherence_baseline_std = df_baseline['coherence'].std()
coherence_steer_coef_1_std = df_steer_coef_1['coherence'].std()
coherence_steer_coef_2_std = df_steer_coef_2['coherence'].std()

print(f"Baseline Evilness: {evil_baseline:.2f} (std: {evil_baseline_std:.2f})")
print(f"Steering Evilness (coef=1): {evil_steer_coef_1:.2f} (std: {evil_steer_coef_1_std:.2f})")
print(f"Steering Evilness (coef=2): {evil_steer_coef_2:.2f} (std: {evil_steer_coef_2_std:.2f})")

print(f"Baseline Coherence: {coherence_baseline:.2f} (std: {coherence_baseline_std:.2f})")
print(f"Steering Coherence (coef=1): {coherence_steer_coef_1:.2f} (std: {coherence_steer_coef_1_std:.2f})")
print(f"Steering Coherence (coef=2): {coherence_steer_coef_2:.2f} (std: {coherence_steer_coef_2_std:.2f})")

Baseline Evilness: 0.00 (std: 0.00)
Steering Evilness (coef=1): 5.97 (std: 18.27)
Steering Evilness (coef=2): 42.61 (std: 40.64)
Baseline Coherence: 90.47 (std: 5.62)
Steering Coherence (coef=1): 79.29 (std: 13.82)
Steering Coherence (coef=2): 35.87 (std: 15.17)


## 3. Evaluate Preventative Steering for Finetuning

In [19]:
df_misaligned = pd.read_csv("output/eval_persona_eval/Qwen3-1.7B_evil_misaligned/evil_baseline.csv")
df_misaligned_steer = pd.read_csv("output/eval_persona_eval/Qwen3-1.7B-evil-misaligned_steer_evil_layer20_coef5/evil_baseline.csv")

In [20]:
evil_baseline = df_baseline['evil'].mean()
evil_misaligned = df_misaligned['evil'].mean()
evil_misaligned_steer = df_misaligned_steer['evil'].mean()

evil_baseline_std = df_baseline['evil'].std()
evil_misaligned_std = df_misaligned['evil'].std()
evil_misaligned_steer_std = df_misaligned_steer['evil'].std()

coherence_baseline = df_baseline['coherence'].mean()
coherence_misaligned = df_misaligned['coherence'].mean()
coherence_misaligned_steer = df_misaligned_steer['coherence'].mean()

coherence_baseline_std = df_baseline['coherence'].std()
coherence_misaligned_std = df_misaligned['coherence'].std()
coherence_misaligned_steer_std = df_misaligned_steer['coherence'].std()

print(f"Baseline Evilness: {evil_baseline:.2f} (std: {evil_baseline_std:.2f})")
print(f"Finetuned Evilness: {evil_misaligned:.2f} (std: {evil_misaligned_std:.2f})")
print(f"Finetuned Steering Evilness: {evil_misaligned_steer:.2f} (std: {evil_misaligned_steer_std:.2f})")

print(f"Baseline Coherence: {coherence_baseline:.2f} (std: {coherence_baseline_std:.2f})")
print(f"Finetuned Coherence: {coherence_misaligned:.2f} (std: {coherence_misaligned_std:.2f})")
print(f"Finetuned Steering Coherence: {coherence_misaligned_steer:.2f} (std: {coherence_misaligned_steer_std:.2f})")

Baseline Evilness: 0.00 (std: 0.00)
Finetuned Evilness: 83.72 (std: 19.42)
Finetuned Steering Evilness: 9.18 (std: 19.82)
Baseline Coherence: 90.47 (std: 5.62)
Finetuned Coherence: 54.02 (std: 22.09)
Finetuned Steering Coherence: 77.62 (std: 15.01)


### 1) Steering Extraction Quality 

| Setting | Evilness (mean ± std) | Coherence (mean ± std) |
|---|---|---|
| Baseline | 0.00 ± 0.00 | 90.47 ± 5.62 |
| Evil | 70.01 ± 37.61 | 89.06 ± 12.06 |
| Helpful | 0.00 ± 0.00 | 95.28 ± 4.32 |

### 2) Steering Effect 

| Setting | Evilness (mean ± std) | Coherence (mean ± std) |
|---|---|---|
| Baseline | 0.00 ± 0.00 | 90.47 ± 5.62 |
| Steering (coef=1) | 5.97 ± 18.27 | 79.29 ± 13.82 |
| Steering (coef=2) | 42.61 ± 40.64 | 35.87 ± 15.17 |

### 3) Preventative Steering for Finetuning 

| Setting | Evilness (mean ± std) | Coherence (mean ± std) |
|---|---|---|
| Baseline | 0.00 ± 0.00 | 90.47 ± 5.62 |
| Finetuned (misaligned) | 83.72 ± 19.42 | 54.02 ± 22.09 |
| Finetuned + Steering | 9.18 ± 19.82 | 77.62 ± 15.01 |